source/Nevermore/Transient/DbCommandExtensions.cs

```diff
@@ -83,18 +57,8 @@ public static async Task<DbDataReader> ExecuteReaderWithRetryAsync(this DbComman
                     (commandRetryPolicy ?? RetryPolicy.NoRetry).LoggingRetries(operationName);
                 return await effectiveCommandRetryPolicy.ExecuteActionAsync(async () =>
                 {
-                    var weOwnTheConnectionLifetime = EnsureValidConnection(command, connectionRetryPolicy);
-                    try
-                    {
-                        return await command.ExecuteReaderAsync(commandBehavior, cancellationToken);
-                    }
-                    catch (Exception)
-                    {
-                        if (weOwnTheConnectionLifetime && command.Connection != null &&
-                            command.Connection.State == ConnectionState.Open)
-                            await command.Connection.CloseAsync();
```

Comment:
Shouldn't this close code have been in a finally block, not a catch block???  Could this be keeping our connections open when this method was used, we opened the connection and there wasn't an error?

Reply:
yeah, that looks a bit erroneous :|

Reply:
If you close the connection the reader won't work anymore will it?

Reply:
Probably not... I'll check, but then I'm not sure when the connection would be closed either... following it back to the Stream method (which doesn't return the DbReader object, but a stream of the objects) it doesn't close it either... so I'm hoping that these methods don't own the connection (ever)... perhaps we should ensure that is always the case for these methods?

Reply:
I have reversed these changes, as they were tangengential to the main focus of the task - the use of the cancellationToken, and returned to the old pattern, as I really don't want to break the DbDataReader methods.